### PR TITLE
refactor: restructure policy-decision-point to use internal package paths

### LINKED ISSUE #481: Restructure policy-decision-point: move v1/ packages under internal/
### Current Limitation

The `exchange/policy-decision-point` module places its application code under a public `v1/` directory (`v1/handler.go`, `v1/database.go`, `v1/services`, `v1/models`, etc.), with the root application code living in `package v1`. These packages are implementation details of the policy-decision-point binary and are not intended to be imported by other modules, but the current layout does not enforce this — any module could import them. This is the same limitation that was addressed for `consent-engine` in #479.

### Suggested Improvement

Restructure the module to use Go's `internal/` convention, mirroring the consent-engine change:

- Move `v1/*` (`database`, `handler`, `services`, `models`, `testhelpers`, test helpers) to `internal/*` so the compiler enforces that they remain private to the policy-decision-point module.
- Rename the root `package v1` to `package internal` and update `main.go` to import `.../policy-decision-point/internal`.
- Update all import paths from `.../policy-decision-point/v1/...` to `.../policy-decision-point/internal/...`.

Also as part of the cleanup:

- Update the stale path reference in `docs/DESIGN-esignet-citizen-authentication.md` (points at `policy-decision-point/v1/services/policy_metadata_service.go`).
- Untrack the generated `coverage.html` / `coverage.out` files (they still reference the old `v1/` paths) and add them to `.gitignore`.

### Version

commit 96332355 (branch: main)

### Additional Context

Verified locally that the restructure is safe: `go build ./...`, `go vet ./...`, and `go test ./...` all pass; there are no external importers of the moved packages. Companion to the consent-engine restructure (#479 / PR #480).

### PR DESCRIPTION
**Pull request by mushrafmim**
## Summary

Restructures the `exchange/policy-decision-point` module to use Go's `internal/` convention. The application code previously lived under a public `v1/` directory (with the root app code in `package v1`), which did not prevent other modules from importing what are really implementation details of the policy-decision-point binary. Moving everything under `internal/` lets the compiler enforce that these packages stay private. This mirrors the consent-engine restructure (#479 / PR #480).

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [x] Refactoring (no functional changes)
- [ ] Performance improvement
- [ ] Other (please describe):

## Changes Made

- Moved `policy-decision-point/v1/*` (`database`, `handler`, `services`, `models`, `testhelpers`, and test helpers) to `policy-decision-point/internal/*`.
- Renamed the root `package v1` to `package internal` and updated `main.go` to import `.../policy-decision-point/internal` (calls now read `internal.NewHandler`, `internal.ConnectGormDB`, etc.).
- Updated all import paths from `.../policy-decision-point/v1/...` to `.../policy-decision-point/internal/...`.
- No functional/behavioral changes.

## Testing

- [x] I have tested this change locally
- [ ] I have added unit tests for new functionality
- [x] I have tested edge cases
- [x] All existing tests pass

`go build ./...`, `go vet ./...`, and `go test ./...` all pass (internal, models, services, testhelpers packages green). Verified there are no external importers of the moved packages.

## Checklist

- [x] My code follows the project's style guidelines
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have checked that there are no merge conflicts

## Related Issues

Closes #481

## Additional Notes

Purely a package restructure — no logic changed. Follow-ups not included in this PR: the stale path reference in `docs/DESIGN-esignet-citizen-authentication.md` (points at `policy-decision-point/v1/services/policy_metadata_service.go`), and the tracked `coverage.html` / `coverage.out` files, which still reference the old `v1/` paths and could be untracked + gitignored.

### CODE REVIEW

File: exchange/policy-decision-point/internal/database.go [hunk 1]
```diff
@@ -1,4 +1,4 @@
-package v1
+package internal
```
[the remark this answers is not in the document]
Comment (mushrafmim) [reply exchange/policy-decision-point/internal/database.go]: addressed

File: exchange/policy-decision-point/internal/handler.go [hunk 2]
```diff
@@ -1,12 +1,12 @@
-package v1
+package internal
```
[the remark this answers is not in the document]
Comment (mushrafmim) [reply exchange/policy-decision-point/internal/handler.go]: addressed

File: exchange/policy-decision-point/internal/test_helpers.go [hunk 3]
```diff
@@ -1,4 +1,4 @@
-package v1
+package internal
```
[the remark this answers is not in the document]
Comment (mushrafmim) [reply exchange/policy-decision-point/internal/test_helpers.go]: addressed

File: exchange/policy-decision-point/main.go [hunk 4]
```diff
@@ -71,7 +71,7 @@ func main() {
 	}()
 
 	// Initialize V1 handlers
-	v1Handler := v1.NewHandler(gormDB)
+	v1Handler := internal.NewHandler(gormDB)
```
[the remark this answers is not in the document]
Comment (mushrafmim) [reply exchange/policy-decision-point/main.go]: addressed

**Review by ginaxu1 — APPROVED** [review]
Lgtm
